### PR TITLE
Fixed bugs which blocked bosh deployment under special condition

### PR DESCRIPTION
**Pull request by jianqiu**
1. has_vm should not return error when no vm found.
2. delete_vm should not return error when no vm found
3. Updated godep workspaces to including new create_disk method of softlayer-go
4. http registry support

### CODE REVIEW

File: action/has_vm.go [hunk 1]
```diff
@@ -20,7 +18,7 @@ func NewHasVM(
 func (a HasVMAction) Run(vmCID VMCID) (bool, error) {
 	_, found, err := a.vmFinder.Find(int(vmCID))
 	if err != nil {
-		return false, bosherr.WrapErrorf(err, "Finding VM '%d'", vmCID)
+		return false, nil
```
Comment (maximilien) [inline action/has_vm.go]: How can we return nil here when previously there was an error?

File: action/delete_vm.go [hunk 2]
```diff
@@ -18,12 +18,7 @@ func NewDeleteVM(
 }
 
 func (a DeleteVMAction) Run(vmCID VMCID) (interface{}, error) {
-	vm, found, err := a.vmFinder.Find(int(vmCID))
-
-	if err != nil {
-		return nil, bosherr.WrapErrorf(err, "Finding vm '%s'", vmCID)
-	}
-
+	vm, found, _ := a.vmFinder.Find(int(vmCID))
```
Comment (maximilien) [inline action/delete_vm.go]: Why would we ignore error here?

File: action/has_vm_test.go [hunk 3]
```diff
@@ -50,8 +50,7 @@ var _ = Describe("HasVM", func() {
 				vmFinder.FindErr = errors.New("fake-find-err")
 
 				found, err := action.Run(1234)
-				Expect(err).To(HaveOccurred())
-				Expect(err.Error()).To(ContainSubstring("fake-find-err"))
+				Expect(err).ToNot(HaveOccurred())
```
Comment (maximilien) [inline action/has_vm_test.go]: Whey remove the second check, important to smoke sure correct error is sent

File: dev/config.json [hunk 4]
```diff
@@ -4,7 +4,7 @@
     "properties": {
       "softlayer": {
         "username": "fake-username",
-        "apiKey": "fake-api-key"
+        "apiKey": "fake-apikey"
```
Comment (maximilien) [inline dev/config.json]: Why this change?

File: dev/create_vm.json [hunk 5]
```diff
@@ -34,6 +34,7 @@
     {
       "default": {
         "type": "dynamic",
+        "ip": "10.244.16.18",
```
Comment (maximilien) [inline dev/create_vm.json]: Why is this IP fix?

File: action/has_vm.go (hunk 1 above)
Comment (jianqiu) [inline action/has_vm.go]: CPI should allow VMNotFoundError otherwise it will block the deployment. 
The action is compliance with the specification of CPI API v1
http://bosh.io/docs/cpi-api-v1.html

File: action/delete_vm.go (hunk 2 above)
Comment (jianqiu) [inline action/delete_vm.go]: CPI should allow VMNotFoundError otherwise it will block the deployment. 
The action is compliance with the specification of CPI API v1
http://bosh.io/docs/cpi-api-v1.html

File: action/has_vm_test.go (hunk 3 above)
Comment (jianqiu) [inline action/has_vm_test.go]: CPI should allow VMNotFoundError otherwise it will block the deployment. 
The action is compliance with the specification of CPI API v1
http://bosh.io/docs/cpi-api-v1.html

File: dev/config.json (hunk 4 above)
Comment (jianqiu) [inline dev/config.json]: Compliance with the style of "fake-username"

File: dev/create_vm.json (hunk 5 above)
Comment (jianqiu) [inline dev/create_vm.json]: @maximilien 
For dev testing.

File: action/delete_vm.go (hunk 2 above)
Comment (maximilien) [inline action/delete_vm.go]: So then we should check for `found == true`  but some `err` occurred or not return any error from Find. Thoughts?

File: action/has_vm_test.go (hunk 3 above)
Comment (maximilien) [inline action/has_vm_test.go]: See message above

File: dev/config.json (hunk 4 above)
Comment (maximilien) [inline dev/config.json]: yeah but we use `SL_API_KEY` so we already are separating `API` and `KEY`

File: dev/create_vm.json (hunk 5 above)
Comment (maximilien) [inline dev/create_vm.json]: Hmm, I hate having this in the public code since that's a potential security vulnerability.

**Comment by cppforlife** [comment]
why do we need registry?

File: action/delete_vm.go (hunk 2 above)
Comment (jianqiu) [inline action/delete_vm.go]: vmFinder could avoid this situation.

**Comment by jianqiu** [comment]
@cppforlife 
As you know, aws and openstack cpi used registry to transfer bosh-agent metadata between CPI and bosh-agent, at present, softlayer-cpi used ssh protocol to communicate with deployed jobs, and update its bosh-agent metadata. we took the interactions between the CPI and the Agent on CloudStack as an example, we want to make similiar implementation, the advantage is that it could avoid the ssh interaction between cpi and agent. Any thoughts on it? Thanks!
http://bosh.io/docs/agent-cpi-interactions.html

File: action/delete_vm.go (hunk 2 above)
Comment (maximilien) [inline action/delete_vm.go]: OK, please change.

**Comment by maximilien** [comment]
@jianqiu I see no updates to this, just a response to @cppforlife. Are you working on new PR?

**Comment by maximilien** [comment]
@jianqiu also, feel free to include comments with links to bosh.io CPI docs where needed. Thx.

File: action/delete_vm.go (hunk 2 above)
Comment (jianqiu) [inline action/delete_vm.go]: Find() method had already been changed.

Comment (maximilien) [inline action/delete_vm.go]: OK but then it should not return an error and thus no need to ignore it :)

**Comment by jianqiu** [comment]
@maximilien 
Could you please review the PR again? We really need it to generate new release for testing, Thanks!

**Comment by maximilien** [comment]
@jianqiu I am merging this due to urgency of tests but I am not happy as you have not addressed all my comments. I will add a chore for you address these comments and I will monitor that it gets done. Thx.

**Comment by maximilien** [comment]
Tracker story [126195721](https://www.pivotaltracker.com/story/show/126195721)
